File: app/src/main/java/ykk/cb/com/cbwms/produce/adapter/Prod_WorkBySaoMaFragment1Adapter.java
package ykk.cb.com.cbwms.produce.adapter;

import android.app.Activity;
import android.text.Html;
import android.view.View;
import android.widget.TextView;

import java.text.DecimalFormat;
import java.util.List;

import ykk.cb.com.cbwms.R;
import ykk.cb.com.cbwms.model.WorkRecordSaoMaTemp;
import ykk.cb.com.cbwms.util.basehelper.BaseArrayRecyclerAdapter;

public class Prod_WorkBySaoMaFragment1Adapter extends BaseArrayRecyclerAdapter<WorkRecordSaoMaTemp> {
    private DecimalFormat df = new DecimalFormat("#.######");
    private Activity context;
    private MyCallBack callBack;

    public Prod_WorkBySaoMaFragment1Adapter(Activity context, List<WorkRecordSaoMaTemp> datas) {
        super(datas);
        this.context = context;
    }

    @Override
    public int bindView(int viewtype) {
        return R.layout.prod_work_saoma_fragment1_item1;
    }

    @Override
    public void onBindHoder(RecyclerHolder holder, final WorkRecordSaoMaTemp entity, final int pos) {
        // 初始化id
        TextView tv_row = holder.obtainView(R.id.tv_row);
        TextView tv_prodNo = holder.obtainView(R.id.tv_prodNo);
        TextView tv_mtlName = holder.obtainView(R.id.tv_mtlName);
        TextView tv_realWorkQty = holder.obtainView(R.id.tv_realWorkQty);
        TextView tv_prodQty = holder.obtainView(R.id.tv_prodQty);
        TextView tv_prodSeqNumber = holder.obtainView(R.id.tv_prodSeqNumber);
        TextView tv_delRow = holder.obtainView(R.id.tv_delRow);

        // 赋值
        tv_row.setText(String.valueOf(pos + 1));
        tv_prodNo.setText(entity.getWorkRecordSaoMa().getProdNo());
        tv_mtlName.setText(entity.getWorkRecordSaoMa().getMtlName());
        tv_realWorkQty.setText(Html.fromHtml("<font><small>"+entity.getStrLocaltionQty()+"</small></font>"));
        tv_prodQty.setText(df.format(entity.getWorkRecordSaoMa().getProdQty())+entity.getWorkRecordSaoMa().getUnitName());
        tv_prodSeqNumber.setText(entity.getWorkRecordSaoMa().getProdSeqNumber()+"\n"+entity.getWorkRecordSaoMaEntry1().getBarcode());

        View view = (View) tv_row.getParent();
        if(entity.isCheckRow()) {
            view.setBackgroundResource(R.drawable.back_style_check1_true);
        } else {
            view.setBackgroundResource(R.drawable.back_style_check1_false);
        }

        View.OnClickListener click = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                switch (v.getId()){
                    case R.id.tv_realWorkQty: // 数量
                        if(callBack != null) {
                            callBack.onClick_num(entity, pos);
                        }
                        break;
                    case R.id.tv_delRow: // 删除行
                        if(callBack != null) {
                            callBack.onClick_delRow(entity, pos);
                        }
                        break;
                }
            }
        };
        tv_realWorkQty.setOnClickListener(click);
        tv_delRow.setOnClickListener(click);
    }

    public void setCallBack(MyCallBack callBack) {
        this.callBack = callBack;
    }

    public interface MyCallBack {
        void onClick_num(WorkRecordSaoMaTemp entity, int position);
        void onClick_delRow(WorkRecordSaoMaTemp entity, int position);
    }


}

File: app/src/main/java/ykk/cb/com/cbwms/produce/adapter/Prod_WorkBySaoMaFragment1Adapter.kt
package ykk.cb.com.cbwms.produce.adapter

import android.app.Activity
import android.text.Html
import android.view.View
import android.widget.TextView

import java.text.DecimalFormat

import ykk.cb.com.cbwms.R
import ykk.cb.com.cbwms.model.WorkRecordSaoMaTemp
import ykk.cb.com.cbwms.util.basehelper.BaseArrayRecyclerAdapter
import ykk.cb.com.cbwms.util.basehelper.BaseRecyclerAdapter

class Prod_WorkBySaoMaFragment1Adapter(private val context: Activity, datas: List<WorkRecordSaoMaTemp>) : BaseArrayRecyclerAdapter<WorkRecordSaoMaTemp>(datas) {
    private val df = DecimalFormat("#.######")
    private var callBack: MyCallBack? = null

    override fun bindView(viewtype: Int): Int {
        return R.layout.prod_work_saoma_fragment1_item1
    }

    override fun onBindHoder(holder: BaseRecyclerAdapter.RecyclerHolder, entity: WorkRecordSaoMaTemp, pos: Int) {
        // 初始化id
        val tv_row = holder.obtainView(R.id.tv_row) as TextView
        val tv_prodNo = holder.obtainView(R.id.tv_prodNo) as TextView
        val tv_mtlName = holder.obtainView(R.id.tv_mtlName) as TextView
        val tv_realWorkQty = holder.obtainView(R.id.tv_realWorkQty) as TextView
        val tv_prodQty = holder.obtainView(R.id.tv_prodQty) as TextView
        val tv_prodSeqNumber = holder.obtainView(R.id.tv_prodSeqNumber) as TextView
        val tv_delRow = holder.obtainView(R.id.tv_delRow) as TextView

        // 赋值
        tv_row.text = (pos + 1).toString()
        tv_prodNo.text = entity.workRecordSaoMa.prodNo
        tv_mtlName.text = entity.workRecordSaoMa.mtlName
        tv_realWorkQty.text = Html.fromHtml("<font><small>" + entity.strLocaltionQty + "</small></font>")
        tv_prodQty.text = df.format(entity.workRecordSaoMa.prodQty) + entity.workRecordSaoMa.unitName
        tv_prodSeqNumber.text = entity.workRecordSaoMa.prodSeqNumber + "\n" + entity.workRecordSaoMaEntry1.barcode

//        val view = tv_row!!.getParent() as View
//        if (entity.isCheckRow) {
//            view.setBackgroundResource(R.drawable.back_style_check1_true)
//        } else {
//            view.setBackgroundResource(R.drawable.back_style_check1_false)
//        }

        val click = View.OnClickListener { v ->
            when (v.id) {
                R.id.tv_realWorkQty // 数量
                -> if (callBack != null) {
                    callBack!!.onClick_num(entity, pos)
                }
                R.id.tv_delRow // 删除行
                -> if (callBack != null) {
                    callBack!!.onClick_delRow(entity, pos)
                }
            }
        }
        tv_realWorkQty!!.setOnClickListener(click)
        tv_delRow!!.setOnClickListener(click)
    }

    fun setCallBack(callBack: MyCallBack) {
        this.callBack = callBack
    }

    interface MyCallBack {
        fun onClick_num(entity: WorkRecordSaoMaTemp, position: Int)
        fun onClick_delRow(entity: WorkRecordSaoMaTemp, position: Int)
    }


}
